client.go

```diff
@@ -924,6 +924,55 @@ func (c *Client) MkdirAll(path string) error {
 	return nil
 }
 
+// RemoveAll delete files recursively in the directory and Recursively delete subdirectories.
+// An error will be returned if no file or directory with the specified path exists
+func (c *Client) RemoveAll(path string) error {
+
+	// Get the file/directory information
+	fi, err := c.Stat(path)
+	if err != nil {
+		return err
+	}
+
+	if fi.IsDir() {
+		// Delete files recursively in the directory
+		files, err := c.ReadDir(path)
+		if err != nil {
+			return err
+		}
+
+		for _, file := range files {
+			if file.IsDir() {
+				// Recursively delete subdirectories
+				err = c.RemoveAll(path + "/" + file.Name())
+				if err != nil {
+					return err
+				}
+			} else {
+				// Delete individual files
+				err = c.Remove(path + "/" + file.Name())
+				if err != nil {
+					return err
+				}
+			}
+		}
+
+		// Delete the empty directory
+		err = c.RemoveDirectory(path)
+		if err != nil {
+			return c.RemoveDirectory(path)
+		}
+	} else {
+		// Delete individual files
+		err = c.Remove(path)
+		if err != nil {
+			return c.Remove(path)
+		}
+	}
+
+	return nil
```

Comment:
Previous comment has not been addressed. This whole block should be replaced by `return c.Remove(path)`